packages/styled-components/src/models/StyleTags.js

```diff
@@ -58,7 +58,11 @@ const addUpUntilIndex = (sizes: number[], index: number): number => {
 
 /* create a new style tag after lastEl */
 const makeStyleTag = (target: ?HTMLElement, tagEl: ?Node, insertBefore: ?boolean) => {
-  const el = document.createElement('style');
+  let targetDocument = document;
+  if(target) targetDocument = target.ownerDocument;
+  else if(tagEl) targetDocument = tagEl.ownerDocument;
+
+  const el = targetDocument.createElement('style');
```

Comment:
Interesting that this causes no security issues; apparently any `<style>` tags' associated `StyleSheet` are considered same-origin. https://html.spec.whatwg.org/multipage/semantics.html#the-style-element:concept-css-style-sheet-origin-clean-flag

This can still break for CSP reasons, though, but it'd break regardless of whether we use CSSOM or child text nodes.